react/features/base/dialog/components/SimpleBottomSheet.native.js

```diff
@@ -0,0 +1,189 @@
+import { Icon } from '../../font-icons';
+
+import { bottomSheet as styles } from './styles';
+
+/**
+ * Underlay color for the buttons on the sheet.
+ *
+ * @type {string}
+ */
+const BUTTON_UNDERLAY_COLOR = '#eee';
+
+/**
+ * A component emulating Android's BottomSheet, in a simplified form.
+ * It supports text options with an icon, which the user can tap. The style has
+ * been implemented following the Material Design guidelines for bottom
+ * sheets: https://material.io/guidelines/components/bottom-sheets.html
+ *
+ * For all intents and purposes, this component has been designed to work and
+ * behave as a {@code Dialog}.
+ */
+class SimpleBottomSheet extends Component<*> {
+    /**
+     * {@code SimpleBottomSheet}'s React {@code Component} prop types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * Used to hide the sheet on cancel.
+         */
+        dispatch: PropTypes.func,
+
+        /**
+         * Handler for the cancel event, which happens when the user dismisses
+         * the sheet.
+         */
+        onCancel: PropTypes.func,
+
+        /**
+         * Handler for the event when an option has been selected in the sheet.
+         */
+        onSubmit: PropTypes.func,
+
+        /**
+         *
+         */
+        options: PropTypes.arrayOf(PropTypes.shape({
+            iconName: PropTypes.string,
+            selected: PropTypes.bool,
+            text: PropTypes.string
+        }))
+    };
+
+    /**
+     * Initializes a new {@code SimpleBottomSheet} instance.
+     *
+     * @param {Object} props - The read-only React {@code Component} props with
+     * which the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._onButtonPress = this._onButtonPress.bind(this);
+        this._onCancel = this._onCancel.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <Modal
+                animationType = { 'slide' }
+                onRequestClose = { this._onCancel }
+                transparent = { true }
+                visible = { true }>
+                <View style = { styles.container }>
+                    <TouchableWithoutFeedback
+                        onPress = { this._onCancel } >
+                        <View style = { styles.overlay } />
+                    </TouchableWithoutFeedback>
+                    <View style = { styles.sheet }>
+                        <View style = { styles.rowsWrapper }>
+                            { this._renderOptions() }
+                        </View>
+                    </View>
+                </View>
+            </Modal>
+        );
+    }
+
+    /**
+     * Dispatches a redux action to hide this sheet.
+     *
+     * @returns {void}
+     */
+    _hide() {
+        this.props.dispatch(hideDialog());
```

Comment:
I'd prefer it if you're able to `hideDialog` with an argument, not just about any dialog.

Reply:
Ack.